packages/core/src/rules/arazzo/strict-source-description-type.ts

```diff
@@ -0,0 +1,19 @@
+import type { ArazzoRule } from '../../visitors';
+import type { UserContext } from '../../walk';
+
+export const StrictSourceDescriptionType: ArazzoRule = () => {
+  return {
+    SourceDescriptions: {
+      enter(SourceDescriptions, { report, location }: UserContext) {
+        SourceDescriptions.forEach((sourceDescription: any) => {
+          if(!['openapi', 'arazzo'].includes(sourceDescription?.type)) {
+            report({
+              message: 'The `type` property of the `sourceDescription` object must be either `openapi` or `arazzo`.',
+              location: location.key(),
```

Comment:
@tatomyr is there the way to provide better location for list elements?
e.g. SourceDescriptions is a list of 3 different type of sourceDescription and I would like to highlight Nth element `type` field. I found a child property but not sure how to select specific item.

```
const SourceDescriptions: NodeType = {
  properties: {},
  items: (value: any) => {
    if (value?.type === 'openapi') {
      return 'OpenAPISourceDescription';
    } else if (value?.type === 'arazzo') {
      return 'ArazzoSourceDescription';
    } else {
      return 'NoneSourceDescription';
    }
  },
};
```

Reply:
found it